dev/Repeater/ViewportManagerWithPlatformFeatures.cpp

```diff
@@ -264,6 +279,35 @@ void ViewportManagerWithPlatformFeatures::OnEffectiveViewportChanged(winrt::Fram
     REPEATER_TRACE_INFO(L"%ls: \tEffectiveViewportChanged event callback \n", GetLayoutId().data());
     UpdateViewport(args.EffectiveViewport());
 
+    // As long as the make_anchor element is within the visible window, force it as the anchor.
+    // As soon as it goes outside, stop forcing it to be the anchor and
+    // let the anchor provider pick a new anchor.
+    if (auto makeAnchorElement = m_makeAnchorElement.get())
+    {
+        auto visibleWindow = GetLayoutVisibleWindowDiscardAnchor();
+        auto layoutSlot = winrt::LayoutInformation::GetLayoutSlot(makeAnchorElement.as<winrt::FrameworkElement>());
+        auto adjustedLayoutSlot = winrt::Rect(layoutSlot.X + m_layoutExtent.X, layoutSlot.Y + m_layoutExtent.Y, layoutSlot.Width, layoutSlot.Height);
+        bool doesNotOverlap = (adjustedLayoutSlot.X + adjustedLayoutSlot.Width) < visibleWindow.X ||
+            (visibleWindow.X + visibleWindow.Width) < adjustedLayoutSlot.X ||
+            (adjustedLayoutSlot.Y + adjustedLayoutSlot.Height) < visibleWindow.Y ||
+            (visibleWindow.Y + visibleWindow.Height) < adjustedLayoutSlot.Y;
+        if (doesNotOverlap)
+        {
+            m_makeAnchorElement.set(nullptr);
+
+            // Now that the anchor is gone, any item can be the anchor.
+            auto children = m_owner->Children();
+            for (unsigned i = 0u; i < children.Size(); ++i)
```

Comment:
Does the Children() enumeration type support a ranged for?
```
for (auto& child : m_owner->Children())

Reply:
nit: prefer `const auto&` or `auto&&` (auto works too, but incurs a copy).